core/src/main/java/com/linecorp/armeria/internal/common/util/ServiceNamingUtil.java

```diff
@@ -51,5 +54,9 @@ public static String fullTypeHttpServiceName(HttpService service) {
         }
     }
 
+    public static String trimTrailingDollarSigns(String serviceName) {
+        return TRAILING_DOLLARS.matcher(serviceName).replaceFirst("");
```

Comment:
Could you run a benchmark to compare the performance between regex and a loop?

Reply:
Sure. I will share after testing.

Reply:
Regex is over 20 times slower than looping. I revert not to use regex. Thanks.

| Benchmark | Mode | Cnt | Score | Error | Units |
| -- | -- | -- | -- | -- | -- |
| Benchmark.regex | thrpt | 5 | 3272303.381 | ± 30980.844 | ops/s |
| Benchmark.loop | thrpt | 5 | 86939690.178 | ± 9717280.541 | ops/s |

Reply:
Thanks for sharing the result. It is a big difference than I expected.

Reply:
Yeah I also did not expect the big difference as it is. 🤣